plugins/org.jboss.tools.ws.jaxrs.ui/plugin.xml

```diff
@@ -636,5 +636,16 @@
       </perspectiveExtension>
    </extension>
 
-       
+   <extension
+         point="org.eclipse.ui.ide.projectConfigurators">
+      <projectConfigurator
+            class="org.jboss.tools.ws.jaxrs.ui.internal.importer.JaxRsConfigurator"
+            label="Java EE Jax-RS">
+         <activeWhen>
+            <hasFileRecursively
+                  filename="web.xml">

```

Comment:
AFAIK, yes, m2e analyses the dependencies and looks for a specific class (`javax.ws.rs.core.Application`, IIRC) in the jars.